File: src/Scripting/PythonPY/bridge.py
class TvDevice:
    def power_on(self):
        return "TV:on"

    def mute(self):
        return "TV:muted"


class RadioDevice:
    def power_on(self):
        return "Radio:on"

    def mute(self):
        return "Radio:muted"


class BasicRemote:
    def __init__(self, device):
        self.device = device

    def activate(self):
        return self.device.power_on()


class MuteRemote:
    def __init__(self, device):
        self.device = device

    def activate(self):
        return self.device.mute()


tv = TvDevice()
radio = RadioDevice()
print(f"basic-tv={BasicRemote(tv).activate()}")
print(f"basic-radio={BasicRemote(radio).activate()}")
print(f"mute-tv={MuteRemote(tv).activate()}")
print(f"mute-radio={MuteRemote(radio).activate()}")

File: src/Systems/Objective-C/bridge.m
#import <Foundation/Foundation.h>

@protocol BridgeDevice <NSObject>
- (NSString *)powerOn;
- (NSString *)mute;
@end

@interface TvDevice : NSObject <BridgeDevice>
@end
@implementation TvDevice
- (NSString *)powerOn { return @"TV:on"; }
- (NSString *)mute { return @"TV:muted"; }
@end

@interface RadioDevice : NSObject <BridgeDevice>
@end
@implementation RadioDevice
- (NSString *)powerOn { return @"Radio:on"; }
- (NSString *)mute { return @"Radio:muted"; }
@end

@interface BasicRemote : NSObject {
    id<BridgeDevice> _device;
}
@property(nonatomic, strong) id<BridgeDevice> device;
- (instancetype)initWithDevice:(id<BridgeDevice>)device;
- (NSString *)activate;
@end
@implementation BasicRemote
@synthesize device = _device;

- (instancetype)initWithDevice:(id<BridgeDevice>)device {
    self = [super init];
    if (self) { self.device = device; }
    return self;
}
- (NSString *)activate { return [self.device powerOn]; }
@end

@interface MuteRemote : NSObject {
    id<BridgeDevice> _device;
}
@property(nonatomic, strong) id<BridgeDevice> device;
- (instancetype)initWithDevice:(id<BridgeDevice>)device;
- (NSString *)activate;
@end
@implementation MuteRemote
@synthesize device = _device;

- (instancetype)initWithDevice:(id<BridgeDevice>)device {
    self = [super init];
    if (self) { self.device = device; }
    return self;
}
- (NSString *)activate { return [self.device mute]; }
@end

int main(void) {
    @autoreleasepool {
        TvDevice *tv = [TvDevice new];
        RadioDevice *radio = [RadioDevice new];
        printf("basic-tv=%s\n", [[[BasicRemote alloc] initWithDevice:tv].activate UTF8String]);
        printf("basic-radio=%s\n", [[[BasicRemote alloc] initWithDevice:radio].activate UTF8String]);
        printf("mute-tv=%s\n", [[[MuteRemote alloc] initWithDevice:tv].activate UTF8String]);
        printf("mute-radio=%s\n", [[[MuteRemote alloc] initWithDevice:radio].activate UTF8String]);
    }
    return 0;
}
